### ISSUE TITLE
Make token-limit configurable as an INSTANCE scoped variable

### ISSUE THREAD

**Issue by morgo** [pingcap/tidb#34583]
## Enhancement

This is forked from https://github.com/pingcap/tidb/issues/33769

For `token-limit` there is a use-case to make it online configurable, but it's not a great fit for `GLOBAL`. This enhancement proposes that we add an instance scoped variable for it.

For the name, I suggest `tidb_max_connection_concurrency` or `tidb_connection_concurrency_limit`.

This is different from the issues in https://github.com/pingcap/tidb/issues/33769 because it is not a breaking change. It can still be configured in the config file as `token-limit` or via the instance scoped variable that we are introducing.

**Comment by CbcWestwolf** [comment]
> @CbcWestwolf would you like to take a look at this?

Sure! Let me do this convertion.

**Comment by CbcWestwolf** [comment]
> It can still be configured in the config file as `token-limit` or via the instance scoped variable that we are introducing.

Should we move the `token-limit` to [instance] section in config file just like what we have done in https://github.com/pingcap/tidb/pull/33279?

**Comment by morgo** [comment]
> > It can still be configured in the config file as `token-limit` or via the instance scoped variable that we are introducing.
> 
> Should we move the `token-limit` to [instance] section in config file just like what we have done in #33279?

Yes. Under `[instance]` it should have the same name as the sysvar. For the sysvar name,  I suggested tidb_max_connection_concurrency or tidb_connection_concurrency_limit. The problem with "token limit" is it is too tied to an implementation, not what the feature does (a cap on concurrent connections that can be executing).
